synchronization/supervized/supervisor.go

```diff
@@ -0,0 +1,72 @@
+package supervized
+
+import (
+	"context"
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/pkg/errors"
+	"runtime"
+	"strings"
+)
+
+type Errorer interface {
+	Error(message string, fields ...*log.Field)
+}
+
+func OneOff(logger Errorer, f func()) {
+	go func() {
+		defer recoverPanics(logger)
+		f()
+	}()
+}
+
+func LongLiving(ctx context.Context, logger Errorer, f func()) {
+	defer recoverPanics(logger)
+	go func() {
+		for {
+			f()
+			select {
```

Comment:
Since `f()` most probably has its own `select` inside, I'm not sure how I feel about the external `select`. Obviously the external `select` isn't allowed to block - that's why the `default` has to be there.. Consider changing to an `if ctx.Err() != nil` which is the non-blocking way to check a context is still not done, it sends a better message of not being allowed to block here.

Reply:
good point!